tensilelite/Tensile/SolutionStructs.py

```diff
@@ -2143,6 +2143,14 @@ def assignDerivedParameters(state):
     if state["StreamK"] != 0:
       state["GlobalSplitU"] = 0 # Cannot enable both Stream-K and GSU
       state["GlobalSplitUAlgorithm"] = "MultipleBuffer" # Set default Algorithm
+      if not (state["ProblemType"]["DataType"].isSingle() or state["ProblemType"]["DataType"].isHalf() or state["ProblemType"]["DataType"].isBFloat16()):
+        reject(state, "Stream-K currently only tested for F32, F16, BF16")
+      if not (state["ProblemType"]["DataTypeA"].isSingle() or state["ProblemType"]["DataTypeA"].isHalf() or state["ProblemType"]["DataTypeA"].isBFloat16()):
+        reject(state, "Stream-K currently only tested for F32, F16, BF16")
+      if not (state["ProblemType"]["DataTypeB"].isSingle() or state["ProblemType"]["DataTypeB"].isHalf() or state["ProblemType"]["DataTypeB"].isBFloat16()):
+        reject(state, "Stream-K currently only tested for F32, F16, BF16")
+      if not (state["ProblemType"]["DestDataType"].isSingle() or state["ProblemType"]["DestDataType"].isHalf() or state["ProblemType"]["DestDataType"].isBFloat16()):
+        reject(state, "Stream-K currently only tested for F32, F16, BF16")
```

Comment:
Maybe say not supported instead of not tested. If you're checking by type you might as well follow our other messages and also include the line type x for datatypeN not supported before your generic message

Reply:
Thanks, got the messages updated